File: src/main/java/com/footverse/order/dto/OrderSummaryResponse.java
package com.footverse.order.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import com.footverse.order.entity.OrderStatus;
import com.footverse.order.entity.PaymentStatus;

/**
 * A compact order for the caller's order-history list (dto-spec §15). The {@code itemCount} is the
 * sum of the order's item quantities — an aggregate assembled by the service, not stored on the
 * order.
 *
 * @param id            the order id
 * @param orderCode     the immutable human-readable order code
 * @param status        the current order status
 * @param paymentStatus the current payment status
 * @param total         the snapshotted order total
 * @param itemCount     the sum of the order-item quantities
 * @param createdAt     the creation timestamp
 */
public record OrderSummaryResponse(
        Long id,
        String orderCode,
        OrderStatus status,
        PaymentStatus paymentStatus,
        BigDecimal total,
        int itemCount,
        LocalDateTime createdAt) {
}

File: app/lib/features/order/models/order_summary_response.dart
import 'package:json_annotation/json_annotation.dart';

import '../../product/models/decimal_json.dart';
import 'order_status.dart';
import 'payment_status.dart';

part 'order_summary_response.g.dart';

/// A compact order for the caller's order-history list (dto-spec §15).
///
/// [total] is the snapshotted order total, rendered exactly as delivered
/// (dto-spec §1). [itemCount] is Σ of the order-item quantities — an aggregate
/// the server assembles, not the distinct-line count.
@JsonSerializable(createToJson: false)
class OrderSummaryResponse {
  const OrderSummaryResponse({
    required this.id,
    required this.orderCode,
    required this.status,
    required this.paymentStatus,
    required this.total,
    required this.itemCount,
    required this.createdAt,
  });

  factory OrderSummaryResponse.fromJson(Map<String, dynamic> json) =>
      _$OrderSummaryResponseFromJson(json);

  final int id;
  final String orderCode;
  final OrderStatus status;
  final PaymentStatus paymentStatus;
  @JsonKey(fromJson: decimalFromJson)
  final double total;
  final int itemCount;
  final DateTime createdAt;
}
